starter_code/src/movies.js

```diff
@@ -1,19 +1,129 @@
 /* eslint no-restricted-globals: 'off' */
 
 // Iteration 1: All rates average - Get the average of all rates with 2 decimals 
+function ratesAverage(movies) {
+    let ratesTotal = movies.reduce(function (accumulator, currentValue) {
+        if (!currentValue.rate) {
+            return accumulator + 0;
+        }
+        return accumulator + parseInt(currentValue.rate, 10);
+    }, 0);
+    let ratesAv = ratesTotal / movies.length;
+    return Math.round(ratesAv * 100) / 100;
+}
 
- 
 // Iteration 2: Drama movies - Get the average of Drama Movies
-
+function dramaMoviesRate(movies) {
+    dramaMovies = movies.filter( // Need to make movies a new array of JUST drama objects/movies
+        function (movie) {
+            if (movie.genre.indexOf("Drama") !== -1) {
+                return true;
+            }
+        }
+    );
+    if (!dramaMovies.length) return 0;
+    return ratesAverage(dramaMovies);
+}
 
 // Iteration 3: Ordering by duration - Order by time duration, ascending (in growing order)
+function orderByYear(movies) {
+    sortedMovies = movies.sort(
+        function (a, b) {
+            if (a.year !== b.year) {
+                return a.year - b.year;
+            } else {
+                return a.title.localeCompare(b.title);
+            }
+        }
+    );
+    return sortedMovies;
+}
 
-
-// Iteration 4: Steven Spielberg. The best? - How many movies did STEVEN SPIELBERG direct
-
+// Iteration 4: Steven Spielberg. The best? - How many movies did STEVEN SPIELBERG direct (that are dramas)
+function howManyMovies(movies) {
+    let stevenMovies = movies.filter(function (movie) {
+        return ((movie.director === "Steven Spielberg") && (movie.genre.includes("Drama")))
+    })
+    return stevenMovies.length;
+}
 
 // Iteration 5: Alphabetic Order - Order by title and print the first 20 titles
+function orderAlphabetically(movies) {
+    sortedTitles = movies.sort(
+        function (a, b) {
+            return a.title.localeCompare(b.title);
+        }
+    )
+    let sortedTwenty = sortedTitles.slice(0, 20);
+    let finalTwenty = [];
+    sortedTwenty.forEach(
+        function (titles) {
+            finalTwenty.push(titles.title)
+        }
+    )
+    return finalTwenty;
+}
 
```

Comment:
Instead of creating an ampty array funalTwenty and push each title inside, you couls have map through sortedTwenty : 
`
let finalTwenty = sortedTwenty.map(function(movie){ return movie.title }) `